### ISSUE TITLE
CA2000 false positive

### ISSUE THREAD

**Issue by NextTurn** [dotnet/roslyn-analyzers#3208]
~#### Analyzer package~

~[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)~

#### Package Version

v2.9.8 (Latest)

#### Diagnostic ID

[CA2000](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2000)

#### Repro steps

```cs
using System;
using System.IO;

public sealed class Program : IDisposable
{
    public Program(Stream stream)
    {
        if (stream is null)
        {
            throw new ArgumentNullException(nameof(stream));
        }
    }

    public static void Main()
    {
        foreach (object _ in new object[1])
        {
            using FileStream stream = File.OpenRead(string.Empty);
            //                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~
            using Program pe = new Program(Stream.Null);
        }
    }

    public void Dispose() { }
}
```

#### Expected behavior

CA2000 is not reported.

#### Actual behavior

CA2000 is reported for `File.OpenRead`.

**Comment by Bouke** [comment]
This is still happening with the .NET 6 SDK (NETAnalyzers).
